### PR TITLE
Add support to byte array decimal fields

### PR DESCRIPTION
**Pull request by YECHUNAN**
Problem
=======
Address #91

Solution
========
When encountering such byte array represented "Decimal" fields, parse them into raw buffers.

Change summary:
---------------
- Added code to parse "Decimal" type fields represented by byte arrays (fixed length or non-fixed length) into raw buffer values for further client side processing.
- Added two test cases verifying the added code.
- Loosen the precision check to allow values greater than 18 for byte array represented "Decimal" fields.

Steps to Verify:
----------------
- Use the library to open a parquet file which contains a "Decimal" field represented by a byte array whose precision is greater than 18.
- Before the change, library will throw an error saying precision cannot be greater than 18.
- After the change, library will parse those fields to their raw buffer values and return records normally.

### CODE REVIEW

**Review by wilwade — APPROVED** [review]
@YECHUNAN This looks great! While running locally I found a completely separate issue in the CI and build, so pushed up a fix for that, but it has nothing to do with your PR.

Thanks so much!

File: esbuild-serve.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ require('esbuild')
       }, {
         entryPoints: ['parquet.ts'],
         outfile: 'main.js',
-        define: {"process.env.NODE_DEBUG": false, "process.env.NODE_ENV": "\"production\"", global: "window" },
+        define: {"process.env.NODE_DEBUG": "false", "process.env.NODE_ENV": "\"production\"", global: "window" },
```
Comment (wilwade) [inline esbuild-serve.js:14]: esbuild required.

File: esbuild.js [hunk 2]
```diff
@@ -61,6 +61,7 @@ Promise.all(targets.map(esbuild.build))
     })
     .catch(e => {
         console.error("Finished with errors: ", e.toString());
+        process.exit(1);
```
Comment (wilwade) [inline esbuild.js:64]: esbuild failure should be reported as such to stop CI

File: package.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
     "buffer": "^6.0.3",
     "chai": "4.3.6",
     "core-js": "^3.22.5",
-    "esbuild": "^0.14.47",
+    "esbuild": "^0.19.2",
```
Comment (wilwade) [inline package.json:45]: I did this to update esbuild to support the newer ts syntax used.

**Review by shannonwells — APPROVED** [review]
Ran the tests and verified the two test files.  Thanks for this work.

**Pull request merged**